platform/m6/sub/101a/boot.c

```diff
@@ -0,0 +1,251 @@
+#include "cache.h"
+#include "core.h"
+
+#include "../../../generic/mmu_utils.h"
+#include "boot.h"
+
+#define CHDK_MMUTBL_START 0xdffda400
+#define CHDK_PAGES_START  0x11d0000
+#define CANON_TABLES_AT   0xdffc0000
+
+static int no_chdk_please = 0;
+
+void CreateTask_spytask() {
+	if (!no_chdk_please) {
+		_CreateTask("SpyTask", 0x19, 0x2000, core_spytask, 0);
+	}
+}
+
+void startup_mode_fix(void) {
+	int startupmode = *(int*) 0x9eb0;
+	switch (startupmode) {
+	case 0x80000:
+	case 0x400000:
+	case 0x200000:
+		no_chdk_please = 1;
+		break;
```

Comment:
done this and CHDK is now loaded

Reply:
The delay is due to the missing hook in init_file_modules_task.